itest/lnd_onchain_test.go

```diff
@@ -278,9 +278,14 @@ func runCPFP(ht *lntest.HarnessTest, alice, bob *node.HarnessNode) {
 		Outpoint: op,
 		// We use a higher fee rate than the default max and expect the
 		// sweeper to cap the fee rate at the max value.
-		SatPerVbyte: maxFeeRate * 2,
+		SatPerVbyte:  maxFeeRate * 2,
+		IncludeRawTx: true,
 	}
-	bob.RPC.BumpFee(bumpFeeReq)
+
+	go func() {
+		bumpFeeResp := bob.RPC.BumpFee(bumpFeeReq)
+		require.NotEmpty(ht, bumpFeeResp.SweepTxHex)
+	}()
```

Comment:
what is the reason that we need to spin this off in a goroutine? is it because BumpFee blocks until the tx is included in a block (meaning we need the "mine blocks" command below to run)? 

I think it is important that we know exactly why this blocks. Also - if it is blocking until the tx is mined, we should determine if this is in fact the behaviour that we want here. Is it possible to get the channel to return the raw hex at the time of tx construction rather? or do we wait until a block confirmation incase we update the structure of the tx? 

Regarding the goroutine - if we do decide that we in fact need to use a goroutine here, we should never do it without tracking the go routine - as it currently stands, we have no idea if the code in here ever even gets run or that it completes. So you would need to add a wait group or something so that we can wait on it later.

Reply:
> What is the reason that we need to spin this off in a goroutine? is it because BumpFee blocks until the tx is included in a block (meaning we need the "mine blocks" command below to run)?

Yes, that was my intent!

Reply:
> Regarding the goroutine - if we do decide that we in fact need to use a goroutine here, we should never do it without tracking the go routine - as it currently stands, we have no idea if the code in here ever even gets run or that it completes. So you would need to add a wait group or something so that we can wait on it later.

Alright 👍

Reply:
> Is it possible to get the channel to return the raw hex at the time of tx construction rather?

Since the `resultChan` is not accessible in [sweep/sweeper.go/createSweepTx](https://github.com/lightningnetwork/lnd/blob/5599b3c9e2fa9dade884e1a6de96479c9697e35a/sweep/sweeper.go#L1166-L1169) where we create sweep transaction may be not feasible to get the raw hex at the time of tx construction ?

Reply:
> do we wait until a block confirmation incase we update the structure of the tx?

 I don't see any mutation for the structure of spent sweep transaction after the transaction confirmation (monitored by `monitorSpend` function) in [handleInputSpent](https://github.com/lightningnetwork/lnd/blob/5599b3c9e2fa9dade884e1a6de96479c9697e35a/sweep/sweeper.go#L1572).